File: app/src/main/java/com/falcon/restaurants/screens/restaurant/FilterRestaurantsUseCase.java
package com.falcon.restaurants.screens.restaurant;

import com.falcon.restaurants.base.FilterInterface;
import com.falcon.restaurants.room.restaurant.Restaurant;

import javax.inject.Inject;

public class FilterRestaurantsUseCase extends FilterInterface<Restaurant> {

    @Inject
    public FilterRestaurantsUseCase() {}

    @Override
    public boolean isQueryFound(String query, Restaurant current) {
        return current.getName().toLowerCase().contains(query.toLowerCase());
    }
}

File: app/src/main/java/com/falcon/restaurants/screens/restaurant/FilterRestaurantsUseCase.kt
package com.falcon.restaurants.screens.restaurant

import com.falcon.restaurants.base.FilterBase
import com.falcon.restaurants.room.restaurant.Restaurant

import javax.inject.Inject

class FilterRestaurantsUseCase @Inject constructor(): FilterBase<Restaurant>() {

    override fun isQueryFound(query: String, current: Restaurant): Boolean {
        return current.name.lowercase().contains(query.lowercase())
    }
}
